### PR TITLE
Doracleupdate

### PR DESCRIPTION
**Pull request by Amxx**

### CODE REVIEW

File: key-concepts/proof-of-contribution.md [hunk 1]
```diff
@@ -55,62 +55,66 @@ Below are the details of the implementations:
...
+  In computer science, a deterministic algorithm is an algorithm which, given a particular input, will always produce the same output.
 
-   If a TEE was used to produce the result, the enclave should produce a `/iexec_out/enclaveSig.iexec` that contains the enclave signature \(of the resultHash and resultSeal\). Finally, if the requester asked for a callback, the value of this callback must be specified in `/iexec_out/callback.iexec`. Otherwize, the digest will be sent through the callback.
+  Both the `digest`, the `hash` and the `seal` are automatically computed based on the output of the application. If the output is not entierly deterministic, then the application can specify a deterministic file that should be used for building consensus. In order to do so, the application just as to prodive the path to the deterministic file using a specific entry `deterministic-output-path` in `${IEXEC_OUT}/computed.json`.
```
Comment (james-toussaint) [inline key-concepts/proof-of-contribution.md]:
```suggestion
  Both the `digest`, the `hash` and the `seal` are automatically computed based on the output of the application. If the output is not entierly deterministic, then the application can specify a deterministic file that should be used for building consensus. In order to do so, the application just has to provide the path to the deterministic file using a specific entry `deterministic-output-path` in `${IEXEC_OUT}/computed.json`.
```

File: key-concepts/proof-of-contribution.md [hunk 2]
```diff
@@ -55,62 +55,66 @@ Below are the details of the implementations:
...
+
+  Alternativelly, if the application is used in a doracle contex (the results are designed to be processed on-chain by receiver smart-contracts), then the value of this callback must be specified in `${IEXEC_OUT}/computed.json` under the entry `callback-data`.
```
Comment (james-toussaint) [inline key-concepts/proof-of-contribution.md]:
```suggestion
  Alternativelly, if the application is used in a doracle context (the results are designed to be processed on-chain by receiver smart-contracts), then the value of this callback must be specified in `${IEXEC_OUT}/computed.json` under the entry `callback-data`.
```

File: use-cases/iexec-doracle.md [hunk 3]
```diff
@@ -45,7 +45,14 @@ An iExec dOracle can be seen as an “on-chain API”: fundamentally it is a sim
...
+The off-chain part of a dOracle is a classical iExec dApp, that will be executed on the iExec platform and be replicated on several workers as part of an iExec computation deal. It contains the oracle logic, for example to query a web API and process the result. Whenever an operator wishes to update the dOracle, it requests a computation like in a normal iExec deal, specifying the dOracle app as dApp, and the parameters if applicable. The dOracle result is written in the `${IEXEC_OUT}/computed.json` file by the dApp, under the `callback_data` name.
```
Comment (PierreJeanjacquot) [inline use-cases/iexec-doracle.md]: under the `callback_data` name.
```suggestion
The off-chain part of a dOracle is a classical iExec dApp, that will be executed on the iExec platform and be replicated on several workers as part of an iExec computation deal. It contains the oracle logic, for example to query a web API and process the result. Whenever an operator wishes to update the dOracle, it requests a computation like in a normal iExec deal, specifying the dOracle app as dApp, and the parameters if applicable. The dOracle result is written in the `${IEXEC_OUT}/computed.json` file by the dApp, under the `callback_data` key.
```

File: use-cases/iexec-doracle.md [hunk 4]
```diff
@@ -57,17 +64,16 @@ A simple example of dOracle is available on Github. The following section goes t
...
+The PriceFeed dApp is a simple Node.js script, available at [Kaiko PriceFeed Source](https://github.com/iExecBlockchainComputing/iexec-apps/blob/master/offchain-computing/offchain-tee-kaiko-pricefeed/src/app.py). Given a set of parameters, the application encodes its result so that it can be interpreted by the corresponding dOracle smart contract, stores it in `${IEXEC_OUT}/computed.json`, and stores the hash of this encoded value to perform the consensus. The Worker will then send these values on-chain as part of the task finalization, where they will be accessible by the dOracle smart contract.
 
 For example, given the parameters `"BTC USD 9 2019-04-11T13:08:32.605Z"` the price-oracle application will:
 
 * Retrieve the price of BTC in USD at 2019-04-11T13:08:32.605Z
 * Multiply this value by `10e9` \(to capture the price value more accurately as it will be represented by an integer onchain\)
 * Encode the date, the description \(`"btc-usd-9"`\) and the value using `abi.encode`
-* Store this result in `/iexec_out/callback.iexec`
-* Hash the result and store it in `/iexec_out/determinism.iexec`
+* Store this result in `${IEXEC_OUT}/computed.json` under the `callback-data` name
```
Comment (PierreJeanjacquot) [inline use-cases/iexec-doracle.md]:
```suggestion
* Store this result in `${IEXEC_OUT}/computed.json` under the `callback-data` key
```

File: use-cases/iexec-doracle.md [hunk 5]
```diff
@@ -110,7 +115,7 @@ function _iexecDoracleUpdateSettings(
 internal
 ```
 
-The update function, that takes in input a task id, and reads the `Task` object data from the `IexecHub` smart contract to perform the required checks: that the authorized app, dataset, workerpool, trust level and tags are valid, and that the hash of `results` is equal to the `hashResult` field of the `Task` object \(over which the consensus was reached\). If the task passes the checks then it returns the `results` field of the `Task` object, i.e. the result of the dOracle dApp computation.
+The update function, that takes in input a task id, and reads the `Task` object data from the `IexecProxy` smart contract to perform the required checks: that the execution is complete and that the authorized app, dataset, workerpool, trust level and tags are valid. The `IexecProxy` already checked that the hash of the `resultsCallback` is equal to the `resultDigest` \(over which the consensus was reached\). If the task passes the checks then it returns the `results` field of the `Task` object, i.e. the result of the dOracle dApp computation.
```
Comment (PierreJeanjacquot) [inline use-cases/iexec-doracle.md]:
```suggestion
The update function, that takes in input a task id, and reads the `Task` object data from the `IexecProxy` smart contract to perform the required checks (the execution must be completed; the app, the dataset, and the workerpool must be authorized; the trust level and tags mus be valid). The `IexecProxy` already checked that the hash of the `resultsCallback` is equal to the `resultDigest` \(over which the consensus was reached\). If the task passes the checks then it returns the `results` field of the `Task` object, i.e. the result of the dOracle dApp computation.
```

**Review by PierreJeanjacquot — COMMENTED** [review]
see suggested changes

**Review by PierreJeanjacquot — APPROVED** [review]
:+1:

**Pull request merged**